gaj23/get-going-gone
# Gaj/feature/traveler test

#### Purpose: 
> Check all that apply.

### What does this PR do? 
> Clearly and concisely describe changes made. 

This PR addressed the Traveler class buildout as well as its associated testing. 

To confirm functionality, as, well, a random Traveler was selected and the greeting and "stats" section were updated accordingly. 

Small additional changes were made to the appearance to take into account the need for a home button when navigating away from the home page. 

<details>
<summary>Previous Branch</summary>
<br>
<img width="1227" alt="Screen Shot 2021-01-16 at 2 11 09 PM" src="https://user-images.githubusercontent.com/68332132/104864091-08f6cd00-5906-11eb-8f93-09404e906307.png">
<img width="326" alt="Screen Shot 2021-01-16 at 3 04 50 PM" src="https://user-images.githubusercontent.com/68332132/104864093-098f6380-5906-11eb-99e3-7ec25981ee8f.png">
</details>
<details>
<summary>Current Branch</summary>
<br>
<img width="382" alt="Screen Shot 2021-01-17 at 7 06 41 PM" src="https://user-images.githubusercontent.com/68332132/104864060-ebc1fe80-5905-11eb-950f-a977c4e78cd3.png">
<img width="954" alt="Screen Shot 2021-01-17 at 7 06 31 PM" src="https://user-images.githubusercontent.com/68332132/104864061-ec5a9500-5905-11eb-92c5-5841f44dcf95.png">
</details>

This PR closes #10 and #12 of issues.

### Problems Encountered & Resulting Solutions  
> State any problems encountered 
> Describe your code changes in detail for reviewers.

When building out Traveler, significant effort was put into how to find current, planned, and past trips from the data. After researching options and trying various patterns, I decided to implement [Luxon](https://moment.github.io/luxon/docs/manual/install.html) rather than moment.js due to the approachability of the code as well as the size of the dependency. Once Luxon was implemented, I found it incredibly easy to compare dates. This function, however, ended up being fairly hefty and could use attention with refactoring. 

### Where should the reviewer start?
> What file(s) and line(s) are where changes are made?

Please see src/Travelers.js and test/travelers-test.js to see those changes. 

For the beginning of the domUpdate please see src/domUpdates.js.

Smaller changes were made inside of src/css 
- for logo changes see src/css/base.scss line 21

Additionally, see those reflected structural changes in src/index.html line 15 - 19.

### How should this be tested?
> What steps need to be taken to confirm functionality? 

First please download the dependency: `npm install --save luxon` 

Then, please fun npm test test/traveler-test.js to confirm all tests pass.

### Steps To Take
> Any questions and possible steps moving forward.

Dom manipulation so that the appearance of a specific travelers data occurs in a semi-pleasing manner. Continued building of classes, as well. 

#### Checklist:
